webserver/views/datasets.py

```diff
@@ -92,6 +93,41 @@ def view(id):
     )
 
 
+def get_json(id):
+    dataset = get_dataset(id)
+    dataset_clean = {
+        "name": dataset["name"],
+        "description": dataset["description"],
+        "classes": [],
+        "public": dataset["public"],
+    }
+    for cls in dataset["classes"]:
+        dataset_clean["classes"].append({
+            "name": cls["name"],
+            "description": cls["description"],
+            "recordings": cls["recordings"],
+        })
+
+    return dataset_clean
+
+CSV_DIR = "./code/data/files/csv_files"
+
+@datasets_bp.route("/<uuid:dataset_id>/download")
+def download_csv(dataset_id):
+    # return render_template("datasets/accuracy.html")
+    ds = get_json(dataset_id)
+    ds_csv = ""
+    for ds_class in ds["classes"]:
+        class_name = ds_class["name"]
+        for rec in ds_class["recordings"]:
+            strList = [str(rec), str(class_name)]
+            ds_csv = ds_csv + ",".join(strList) + "\n"
+ 
+    return Response(ds_csv,
+                    mimetype="text/csv",
+                    headers={"Content-disposition":"attachment; filename=dataset.csv"})
```

Comment:
You can use [flask.send_file](http://flask.pocoo.org/docs/1.0/api/#flask.send_file) to send a filepointer as an attachment (the stringio is a filepointer!)
Think about the filename that you send. It should be descriptive, and if I download many datasets I should know which one is which by looking at its filename.